fireant/slicer/widgets/highcharts.py

```diff
@@ -0,0 +1,323 @@
+    def metrics(self):
+        """
+        :return:
+            A set of metrics used in this chart. This collects all metrics across all axes.
+        """
+        seen = set()
+        return [metric
+                for axis in self.axes
+                for metric in axis.metrics
+                if not (metric.key in seen or seen.add(metric.key))]
+
+    def transform(self, data_frame, slicer, dimensions):
+        """
+        - Main entry point -
+
+        Transforms a data frame into highcharts JSON format.
+
+        See https://api.highcharts.com/highcharts/
+
+        :param data_frame:
+            The data frame containing the data. Index must match the dimensions parameter.
+        :param slicer:
+            The slicer that is in use.
+        :param dimensions:
+            A list of dimensions that are being rendered.
+        :return:
+            A dict meant to be dumped as JSON.
+        """
+        colors = itertools.cycle(self.colors)
+
+        levels = list(range(1, len(dimensions)))
+        groups = list(data_frame.groupby(level=levels)) \
+            if levels \
+            else [([], data_frame)]
+
+        dimension_display_values = extract_display_values(dimensions, data_frame)
+        render_series_label = dimensional_metric_label(dimensions, dimension_display_values)
+
+        references = [reference
+                      for dimension in dimensions
+                      for reference in getattr(dimension, 'references', ())]
+
+        y_axes, series = [], []
+        for axis_idx, axis in enumerate(self.axes):
+            colors, series_colors = itertools.tee(colors)
+            axis_color = next(colors) if 1 < len(self.metrics) else None
+
+            # prepend axes, append series, this keeps everything ordered left-to-right
+            y_axes[0:0] = self._render_y_axis(axis_idx,
+                                              axis_color,
+                                              references)
+            series += self._render_series(axis,
+                                          axis_idx,
+                                          axis_color,
+                                          series_colors,
+                                          groups,
+                                          render_series_label,
+                                          references)
+
+        x_axis = self._render_x_axis(data_frame, dimension_display_values)
+        plot_options = self._render_plot_options(data_frame)
+
+        return {
+            "title": {"text": self.title},
+            "xAxis": x_axis,
+            "yAxis": y_axes,
+            "plotOptions": plot_options,
+            "series": series,
+            "tooltip": {"shared": True, "useHTML": True},
+            "legend": {"useHTML": True},
+        }
+
+
+    def _render_x_axis(self, data_frame, dimension_display_values):
+        """
+        Renders the xAxis configuraiton.
+
+        https://api.highcharts.com/highcharts/yAxis
```

Comment:
Good that you're putting the documentation link for the highcharts elements. That helps a lot when debugging or changing behaviour.